modules/substrate/src/verifier.rs

```diff
@@ -176,10 +184,18 @@ where
 		}
 
 		let current_authority_set = self.storage.current_authority_set();
-		let is_finalized = check_finality_proof(&header, &current_authority_set, &proof.0);
-		if !is_finalized {
-			return Err(FinalizationError::UnfinalizedHeader);
-		}
+		let voter_set = VoterSet::new(current_authority_set.authorities).expect(
```

Comment:
(probably add test for this case?)

Reply:
> This proof is invalid - you never check that the set is correct.

Then the proof panicking would've done it's job to indicate a bug :P Yep I'll do it

Reply:
The proof is still incorrect. The fact we have a bug is obvious (since we panicked). The proof should include a bunch of statements that indicate why it won't happen. Something like: `"VoterSet correctness is verified during import, before writing to storage; qed"`